### ISSUE TITLE
Sound Fired events without sound volume

### ISSUE THREAD

**Issue by CreativeTechGuy** [adamk33n3r/runelite-watchdog#63]
Is there a way to have events on sound effects without playing any sound effects?

I know that when the sounds are muted, no events are fired. Would it be possible to have this plugin have an option to listen to, and then `.consume()` every sound played so it doesn't actually make a sound but can still be used to trigger alerts?

**Comment by adamk33n3r** [comment]
I don't think I'd want to consume **every sound played**. So we'd only want to mute sounds the we're triggering on. If we have the config on the alert itself, confusion could arise if you have multiple alerts triggering on the same sound effect and one of them is muted etc. Instead of having it on the alert, we could just have a list of soundids to mute inside the main config. But at that point, might as well use something like https://runelite.net/plugin-hub/show/annoyance-mute. Thoughts?

**Comment by CreativeTechGuy** [comment]
In an ideal world, I'd want the game sounds always muted. But I cannot have both the sounds muted and the sound events fired. So I was hoping there'd be a way to have the sounds on to get the events, but then for this plugin to consume every sound after it fires any relevant events so no sound actually plays. It'd be a global opt-in setting for those who wish to have all sounds muted at all times.

**Comment by adamk33n3r** [comment]
I get the use case, it just doesn't really seam related to Watchdog. I see this as a separate plugin. I don't know much about the "consuming" of sounds. If that just prevents it from playing? Cause if so, there could just be a separate plugin that mutes all sounds and then Watchdog would still be able to trigger on them. I'll think about it, I'm not 100% opposed to adding it to watchdog. But I'm also not opposed to creating a separate companion plugin. Would be nice if the Annoyance Mute plugin had a "mute all" option.

**Comment by CreativeTechGuy** [comment]
I don't think it's technically possible for it to be another plugin, that's the reason why I'm suggesting it here.

So TLDR of consuming events (as far as I understand it):

* Stops event propagation (I don't believe any other plugin or event listener will receive that event after it has been consumed)
  * This is why I don't think it's possible for another plugin to mute all sounds as the plugin event handler registration order would matter and depending on the order, it may or may not work for watchdog notifications.
* Stops the sound from playing. I'm conceptualizing it like the "last" event handler is the one which actually plays the sound, so if any handler consumes it, then it is not played.

The RuneLite "Music" plugin has an option to mute sounds, it does this by calling `.consume()` on a sound event if it matches a certain category. It seems like this still triggers the event in Watchdog, so maybe my point 1 is incorrect. It's hard to test that if it is possibly order dependent.

If this can just be a "mute all sounds" plugin which is separate, then 100% that's a better option. I guess I'll have to make that and try it out. Although still won't be 100% sure if it is order dependent or not.

The best comment I found is [here](https://github.com/runelite/runelite/blob/e3840ca1fadef104ca473fb26029432b3063e855/runelite-api/src/main/java/net/runelite/api/events/GrandExchangeSearched.java#L41-L52) which makes me think that consuming an event will still pass it through all other RuneLite code, but stop it from being passed to the vanilla code? Strange since I'd assume it'd be the other way around, but if that's correct, then a separate plugin is best.

I'll test this out and get back to you. (Although it would be convenient if this was just built-in to Watchdog, :P)

**Comment by adamk33n3r** [comment]
Ya I would expect that it wouldn't consume the events for other plugins to not receive it because like you say, it would be dependent on the order, which you can't specify. But thanks for testing it out!

**Comment by CreativeTechGuy** [comment]
Okay it's definitely better as a separate plugin. I just submitted it to Plugin Hub. [Mute All](https://github.com/CreativeTechGuy/mute-all) if you are curious.

I linked to your plugin in my README as it's likely going to be the best pairing, but there's probably other plugins this will be useful for too.

Appreciate you talking this out with me!

**Comment by adamk33n3r** [comment]
Oh awesome! I was additionally envisioning an optional blacklist where you could put sound id's if you only want to mute some. But that's technically in the Mute Annoyance plugin so I guess if that's wanted, users can just use that one instead haha. Great work!

One thing I noticed is that you have a message show up once for users telling them they need it not muted in game which is cool. I'm wondering if we are able to detect if it's muted or not and then base the message showing on that. In case someone forgets later.

**Comment by CreativeTechGuy** [comment]
> I'm wondering if we are able to detect if it's muted or not and then base the message showing on that

If you can figure this out, please let me know! 99% of the time spent on this was trying to figure that out but I couldn't find a way. Especially since the presence of the `Music` plugin changes the way the settings are handled, so there's two different places that it could be configured (native or overridden Music interface). So I just gave up on that one haha.
